text/2020-01-30-key-spaces.md

```diff
@@ -0,0 +1,181 @@
+# Title
+
+Key Spaces
+
+## Summary
+
+When multiple applications store data in TiKV, we need a clear pattern for avoiding conflicting use of the key space in TiKV.
+TiKV is a monolithic KV store, so each application must choose a key prefix.
+Key Spaces formalizes this process by registering a key space for an application in TiKV itself.
+
+When combined with authentication (not part of this proposal), this can secure an application's key space.
```

Comment:
I' like to know more details about authentication, can you provide an example here?

Reply:
Etcd has the ACL and supports to grant permission for a role on a key prefix. It is an RBAC implementation(IMO). https://github.com/etcd-io/etcd/blob/master/Documentation/op-guide/authentication.md#working-with-roles

Reply:
Besides, [distributedio/titan](https://github.com/distributedio/titan) uses namespace as the key prefix which combines with the authentication feature: User uses a token which is signed by a key with a namespace to connect to titan.

Reply:
Thanks for the etcd link. @shafreeck I would like to find out more about the Titan auth.

Reply:
https://medium.com/@shafreeck/titan-a-distributed-redis-protocol-compatible-nosql-database-4379a1307661

The `Multi-tenancy support` section gives some details.

Reply:
Thanks, that seems similar to what I came up with for my [auth proposal](https://github.com/tikv/rfcs/pull/40)